lightwood/encoders/time_series/rnn.py

```diff
@@ -48,29 +54,39 @@ def to(self, device, available_devices):
         self._encoder = self._encoder.to(self.device)
         return self
 
-    def prepare(self, priming_data, feedback_hoop_function=None, batch_size=256):
+    def prepare(self, priming_data, previous_target_data=None, feedback_hoop_function=None, batch_size=256):
         """
         The usual, run this on the initial training data for the encoder
         :param priming_data: a list of (self._n_dims)-dimensional time series [[dim1_data], ...]
+        :param previous_target_data: tensor with encoded previous target values for autoregressive tasks
         :param feedback_hoop_function: [if you want to get feedback on the training process]
         :param batch_size
         :return:
         """
         if self._prepared:
-            raise Exception('You can only call "prepare_encoder" once for a given encoder.')
+            raise Exception('You can only call "prepare" once for a given encoder.')
         else:
-            self.setup_nn()
+            self.setup_nn(previous_target_data)
+
 
         # Convert to array and determine max length:
         for i in range(len(priming_data)):
             if not isinstance(priming_data[i][0], list):
                 priming_data[i] = [priming_data[i]]  # add dimension for 1D timeseries
-            self._max_ts_length = max(len(priming_data[i][0]), self._max_ts_length)
+            self._max_ts_length = max(len(priming_data[i][0]), self._max_ts_length)  # TODO: this is set at Native... should we still check?
 
+        # normalize data
         if self._normalizer:
             self._normalizer.prepare(priming_data)
 
-        # decrease for small datasets
+        if previous_target_data is not None and len(previous_target_data) > 0:
+            target_dict = previous_target_data[0]
+            normalizer = MinMaxNormalizer()  # TODO: normalizer determined by subtype (for datetime previous_ columns support)
```

Comment:
I think your native PR is almost ready for merging, maybe merge that into `staging` and then add the subtype check here.